homeassistant/components/essent/entity.py

```diff
@@ -0,0 +1,37 @@
+"""Base entity for Essent integration."""
+
+from essent_dynamic_pricing.models import EnergyData
+from homeassistant.helpers.device_registry import DeviceEntryType, DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import ATTRIBUTION, DOMAIN
+from .coordinator import EssentDataUpdateCoordinator
+
+
+class EssentEntity(CoordinatorEntity[EssentDataUpdateCoordinator]):
+    """Base class for Essent entities."""
+
+    _attr_has_entity_name = True
+    _attr_attribution = ATTRIBUTION
+
+    def __init__(
+        self,
+        coordinator: EssentDataUpdateCoordinator,
+        energy_type: str,
+    ) -> None:
+        """Initialize the entity."""
+        super().__init__(coordinator)
+        self.energy_type = energy_type
+        self._attr_device_info = DeviceInfo(
+            entry_type=DeviceEntryType.SERVICE,
+            identifiers={(DOMAIN, coordinator.config_entry.entry_id)},
+            name="Essent",
+            manufacturer="Essent",
+        )
+
+    @property
+    def energy_data(self) -> EnergyData | None:
+        """Return the energy data for this entity."""
+        if (data := self.coordinator.data) is None:
+            return None
```

Comment:
If the data is not there, update failed should've been raised and then the entity is marked as unavailable

Reply:
Ok. I don't know why this matters. i guess the goal is to expose failures instead of masking them? Like a fail fast pattern.
There are many cases where I deal with a potential entity.energy_data is None. I've removed them all.

Reply:
If we can fetch the value but simply don't know, then we should return None. I can't think of any reason why essent would return that. But for example for an air quality sensor if a sensor got disconnected